uimaj-core/src/main/java/org/apache/uima/cas/impl/BinaryCasSerDes.java

```diff
@@ -559,8 +553,10 @@ public SerialFormat reinit(Header h,
       if (!delta) {
         baseCas.resetNoQuestions();
       }
-      
-      
+
+      // isBeforeV3 adjusts binary type numbers, and is reset with the CAS when !delta, so it must be set here.
+      isBeforeV3 = !h.isV3 && h.getSeqVersionNbr() < 2;

```

Comment:
Isn't this line even enough? Do we need to clear the flag in the reset call?

Reply:
Correct, I had this thought myself, but then I thought that it wouldn't formally be a full reset when the flag isn't cleared. You could also argue that what would happen if someone decided to change this line to the way the assignment was done previously without also clearing the flag during reset? The error would be introduced again, so I think it also adds a bit of robustness to these kind of changes, although it's maybe a bit hypothetic. Anyway, that's why I think it is formally better with the extra clear.